pkg/providers/vsphere/vmprovider_vm.go

```diff
@@ -1033,8 +1033,10 @@ func (vs *vSphereVMProvider) vmCreateGenConfigSpecExtraConfig(
 		}
 	}
 
-	// The ConfigSpec's current ExtraConfig values (that came from the class) take precedence over what was set here.
-	createArgs.ConfigSpec.ExtraConfig = util.AppendNewExtraConfigValues(createArgs.ConfigSpec.ExtraConfig, ecMap)
+	// The ConfigSpec's current ExtraConfig values (that came from the class)
+	// take precedence over what was set here.
+	newExtraConfig := pkgutil.OptionValues(ecMap)
+	newExtraConfig.MergeList(createArgs.ConfigSpec.ExtraConfig)
```

Comment:
I see a behavior change from what happened with AppendNewExtraConfigValues(). The earlier method only added new fields from _ecMap_ that didn't exist in _createArgs.ConfigSpec.ExtraConfig_. 

Might need to do something like IIUC
```
createArgsEC := pkgutil.OptionValuesFromList(createArgs.ConfigSpec.ExtraConfig)
newEc := pkgutil.OptionValues(ecMap)
createArgsEc.Append(newEc)
```